api/extension-guides/notebook.md

```diff
@@ -0,0 +1,303 @@
+		cell.metadata.runState = vscode.NotebookCellRunState.Error;
+		cell.metadata.lastRunDuration = undefined;
+	}
+};
+```
+
+#### Output Types
+
+Outputs must be in one of three formats: Text Output, Error Output, or Rich Output. A kernel may provide multiple outputs for a single execution of a cell, in which case they will be displayed as a list.
+
+##### Text Output
+Text outputs are the most simple output format, and work much like many REPL's you may be familar with. They consist only of a `text` field, which is rendered as plain text in the cell's output element:
+```ts
+{
+	outputKind: vscode.CellOutputKind.Text,
+	text: '...'
+}
+```
+![Cell with simple text output](images/notebook/text-output.png)
+
+##### Error Output
+Error outputs are helpful for displaying runtime errors in a consistant and understandable manner. They contain `ename` and `evalue` fields for displaying the error type and message, respectively, as well as `traceback` field which takes a list of strings which get displayed like a callstack:
+```ts
+{
+	outputKind: vscode.CellOutputKind.Error,
+	ename: 'Error Name',
+	evalue: 'Error Value',
+	traceback: ['stack frame 1', 'stack frame 2', 'stack frame 3', 'stack frame 4']
+}
+```
+![Cell with error output showing error name and message, as well as a stack trace](images/notebook/error-output.png)
+
+##### Rich Output
+Rich outputs are the most advanced form of displaying cell outputs. They allow for providing many different representations of the output data, keyed by mimetype. For example, if a cell output was to represent a GitHub Issue the kernel might produce a rich output with several properties on its `data` field:
+- A `text/html` field containing a formatted view of the issue
+- An `application/json` field constiang a machine readable view
+- An `application/github-issue` field that a `NotebookOutputRenderer` could use to create a fully interactive view of the issue
+
+In this case, the `text/html` and `application/json` views will be rendered by VS Code natively, but the `application/github-issue` view will display an error if no `NotebookOutputRenderer` was registered to that mimetype.
+```ts
+{
+	outputKind: vscode.CellOutputKind.Rich,
+	data: {
+		'text/html': '<b>Hello</b> World',
+		'application/json': { hello: 'world' },
+		'application/hello-world': 'my-hello-world-data-interchange-format',
+	}
+}
+```
+![Cell with rich output showing switching betweeen formatted html, a json editor, and an error message showing no renderer is available (application/hello-world)](images/notebook/rich-output.gif)
+
+By default VS Code can render the mimetypes:
+- application/json
+- application/javascript
+- text/html
+- image/svg+xml
+- text/markdown
+- image/png
+- image/jpeg
+- text/plain
+- text/x-javascript
+
+To render an alternative mimetype, a `NotebookOutputRenderer` must be registered for that mimetype.
+
```

Comment:
We may also want to talk about multi-mimetype output and how extensions and users can control the display order of multi-mimetype output.

Reply:
I gave an example above with 
```ts
{
	outputKind: vscode.CellOutputKind.Rich,
	data: {
		'text/html': '<b>Hello</b> World',
		'application/json': { hello: 'world' },
		'application/hello-world': 'my-hello-world-data-interchange-format',
	}
}
```
is that what you mean by multi-mimetype output? 

And how can extensions and users can control the display order of multi-mimetype output?

Reply:
Firstly, we have builtin display orders

https://github.com/microsoft/vscode/blob/d8ef78152ce82d2e821ae2ef85ca08a54f89fa84/src/vs/workbench/contrib/notebook/common/notebookCommon.ts#L33-L53

and both notebook content providers and users can control that: